src/transform.ts

```diff
@@ -5,119 +5,132 @@ import * as JSON5 from "json5";
 const packageName = "env0-ts-transform-json-schema";
 
 export interface TransformerOptions {
-  env: { [key: string]: string };
+    env: { [key: string]: string };
+}
+
+const getFullyQualifiedTypeName = ({ right: node, left: qualifier }) => {
+    const qualifiers = []
+
+    while(node) {
+        qualifiers.push(node.escapedText);
+        // If we don't have right it means we are in the top hierarchy
+        node = qualifier?.right || qualifier;
+        qualifier = qualifier?.left;
```

Comment:
I don't get why we're working on both node and qualifier at the same time.  
My suggestion was to treat the just the same in another iteration

Reply:
without the `|| qualifier` we finish without the top most `namespace` name
then we need to traverse there, and if we don't have `right` we are in the top most node so why traverse there again?

Reply:
https://github.com/env0/ts-transform-json-schema/pull/19/commits/b0b01511f29aa26bb8d316876830f5b2740b22ea